PSKoans/Koans/Constructs and Patterns/AboutRegularExpressions.Koans.ps1

```diff
@@ -0,0 +1,468 @@
+    }
+}
+
+Describe 'Brackets and Braces' {
+    <#
+        Brackets and braces don't have one overarching purpose in regular expressions. Different
+        types of brackets and braces all mean different things. Sometimes other regex symbols
+        meanings change when they're found inside some brackets or braces.
+    #>
+    
+    Context '{ and } - Curly Braces' {
+        It 'works like a custom quantifier' {
+            <#
+                Curly braces are effectively customizable quantifiers. The star, plus, and question
+                mark quantifiers from earlier are static in their meaning. Star always means "zero
+                or more", question mark always means "one or zero", and plus always means "one or
+                more". Those cover tons of use cases and examples, but there are plenty more
+                situations where you want to be more specific.
+            #>
+    
+            <#
+                The portion of this regex in curly braces acts as a quantifier for the symbol that
+                comes immediately before it. It means "exactly 4 digits."
+            #>
+            'I can count to 1024' -match '\d{4}' | Should -BeTrue
+            $matches[0] | Should -Be '____'
+    
+            '____' -match '\w{10}' | Should -BeTrue
+            '____' -match '\w{10}' | Should -BeFalse
+    
+            <#
+                One can use curly braces to give a range of numbers. This one means "between 2 and 4
+                of any character"
+            #>
+            'Paper airplane' -match '.{2,4}'
+    
+            $fruitMatch = 'p____'
+            [regex]::Matches('Apples', $fruitMatch).Value | Should -Be 'pp'
+            $fruitMatch.Contains('}') | Should -BeTrue     # Make sure you use curly braces!
+    
+            <#
+                If you omit the second number, the custom quantifier becomes "this many or more".
+                This one means "two or more non-whitespace characters"
+            #>
+            [bool]$whitespaceMatch = $____ # Should be either $true or $false
+            $whitespaceMatch | Should -Be 'The moon in June is a big balloon' -match '\s{2,}'
+        }
+    }
+
+    Context '( and ) - Round Braces' {
+        It 'groups patterns together' {
+            <#
+                Round brackets are used to create groups of symbols. These groups can then have
+                quantifiers applied to the entire group, or be used in approximately 42 trillion
+                other ways. This is just the tip of the round bracket/regular expression groups
+                iceberg.
+            #>
+    
+            '____' | Should -Be [regex]::Matches('Bears Beat Bongos', '(B.+){3}').Value
+    
+            $badgers = 'BadgerBadgerBadger'
+            $groupingPattern = '____'
+            [regex]::Matches($badgers, $groupingPattern).Value | Should -Be 'BadgerBadgerBadger'
+            $groupingPattern.Contains(')') | Should -BeTrue    # Use a group!
+    
+        }
```

Comment:
Same here too. Pretty sure `.Matches` stores a `.Groups` or something that you can check that more effectively with as well. :)

Reply:
It does, but I don't want to introduce that to the learner :D